### PR TITLE
Fix copy paste #680

### LINKED ISSUE #664: Pasting doesn't work in the mediumjs textfields (e.g. create post)

### PR DESCRIPTION
**Pull request by m0ru**
Test: copy-paste text when creating a need and chatting.

Kinda resolves #664 (see there for the current caveat)

### CODE REVIEW

**Comment by quasarchimaere** [comment]
although i am in favor of some dependency changes (to be exact those dependecies who have been changed to a specific version and not the old version or higher annotation). I still do not like the "regression" that comes with including a version called "master" from github, as it will not be the same within all of our development machines and it could cause problems and regressions in the future when the sources of version "master" changes -> happens for medium.js

therefore i do not want to merge it as is, maybe we can take the currently stable version (the one lying behind master) and reference that version instead of having a wildcard-dependency

**Comment by m0ru** [comment]
The last tagged version is more than a year older that HEAD (that is hasn't changed for half a year now)

File: webofneeds/won-owner-webapp/src/main/webapp/jspm_config.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ System.config({
   },
 
   map: {
-    "Medium.js": "github:jakiestfu/Medium.js@1.0.1",
+    "Medium.js": "github:jakiestfu/Medium.js@master",
```
Comment (quasarchimaere) [inline webofneeds/won-owner-webapp/src/main/webapp/jspm_config.js]: shouldnt this version also be just the frozen on the commit number? see package.json

**Comment by quasarchimaere** [comment]
branch cant be merged, since the fix does not work:


medium.js:270 Uncaught TypeError: Cannot read property 'saveSelection' of undefined
    at Medium.paste (medium.js:270)
    at medium.js:183
    at Object.isModifier (medium.js:1457)
    at HTMLDivElement.down (medium.js:169)
paste @ medium.js:270
(anonymous) @ medium.js:183
isModifier @ medium.js:1457
down @ medium.js:169
medium.js:430 Uncaught TypeError: Cannot read property 'saveSelection' of undefined
    at HTMLDivElement.pasteEventHandler (medium.js:430)

**Comment by m0ru** [comment]
That denotes the folder it puts the dependency into (so if you write `import "Medium.js"` systemjs can resolve that to a proper file path). For some reason all commits (from the master-branch, i guess) use @master as path.

**Comment by quasarchimaere** [comment]
Uncaught TypeError: Cannot read property 'saveSelection' of undefined
    at Medium.paste (medium.js:270)
    at medium.js:183
    at Object.isModifier (medium.js:1457)
    at HTMLDivElement.down (medium.js:169)
medium.js:430 Uncaught TypeError: Cannot read property 'saveSelection' of undefined
    at HTMLDivElement.pasteEventHandler (medium.js:430)
location-picker.js:177 starting type-ahead search for:  test
utils.js:581 About to query nominatim: https://nominatim.openstreetmap.org/search?q=%20test&format=json
medium.js:270 Uncaught TypeError: Cannot read property 'saveSelection' of undefined
    at Medium.paste (medium.js:270)
    at medium.js:183
    at Object.isModifier (medium.js:1457)
    at HTMLDivElement.down (medium.js:169)
medium.js:430 Uncaught TypeError: Cannot read property 'saveSelection' of undefined
    at HTMLDivElement.pasteEventHandler (medium.js:430)
location-picker.js:183 location search results:  Array[10]
medium.js:270 Uncaught TypeError: Cannot read property 'saveSelection' of undefined
    at Medium.paste (medium.js:270)
    at medium.js:183
    at Object.isModifier (medium.js:1457)
    at HTMLDivElement.down (medium.js:169)
medium.js:430 Uncaught TypeError: Cannot read property 'saveSelection' of undefined
    at HTMLDivElement.pasteEventHandler (medium.js:430)
medium.js:270 Uncaught TypeError: Cannot read property 'saveSelection' of undefined
    at Medium.paste (medium.js:270)
    at medium.js:183
    at Object.isModifier (medium.js:1457)
    at HTMLDivElement.down (medium.js:169)


branch can't be merged, copy+paste does not work locally or on integration-test

**Comment by heikofriedrich** [comment]
please resolve merge conflicts

**Comment by m0ru** [comment]
Add comment to `<script>`-tag explaining why it can't be bundled.

**Comment by m0ru** [comment]
Merge conflicts resolved :)

**Comment by heikofriedrich** [comment]
build error on integration test:

[ERROR] npm WARN package.json won-owner-webapp@0.2.0 No description
[ERROR] npm WARN package.json won-owner-webapp@0.2.0 No repository field.
[ERROR] npm WARN package.json won-owner-webapp@0.2.0 No README data
[ERROR] npm ERR! Linux 4.4.0-59-generic
[ERROR] npm ERR! argv "/var/lib/jenkins/workspace/webofneeds-integration-test/webofneeds/won-owner-webapp/src/main/webapp/node/node" "/var/lib/jenkins/workspace/webofneeds-integration-test/webofneeds/won-owner-webapp/src/main/webapp/node/node_modules/npm/bin/npm-cli.js" "install" "--color=false"
[ERROR] npm ERR! node v0.12.0
[ERROR] npm ERR! npm  v2.6.0
[ERROR] 
[ERROR] npm ERR! Unsupported URL Type: github:jakiestfu/Medium.js
[ERROR] npm ERR! 
[ERROR] npm ERR! If you need help, you may report this error at:
[ERROR] npm ERR!     <http://github.com/npm/npm/issues>
[ERROR] 
[ERROR] npm ERR! Please include the following file with any support request:
[ERROR] npm ERR!     /var/lib/jenkins/workspace/webofneeds-integration-test/webofneeds/won-owner-webapp/src/main/webapp/npm-debug.log

**Comment by m0ru** [comment]
Reinstalled everything with the fixed versions and it still works for me. Can we test it on one of your machines next week? :)

**Comment by heikofriedrich** [comment]
you can test either on mine or maybe better on raphaels machine, he just set everything up new and i think its a good starting point for him to get to know things

**Comment by m0ru** [comment]
I updated the npm-version that frontend-maven uses. Stuff should-work-now™.

**Comment by heikofriedrich** [comment]
ok the build was running. I tested the copying in the need view. Two things i recognized:
* after inserting text the cursor is put before the text instead of after which seems odd
* line breaks are not inserted, they seem to be replaced by spaces instead

**Comment by m0ru** [comment]
a) can't be fixed without a lot of effort (i.e. tracking the cursor before pasting and then moving it afterwards), this is the default-behaviour for pasting in contenteditable. chrome has the slightly better default of moving the cursor to the end of the "textfield" (i.e. `<div>`)

I'll take a look at (b).

**Comment by m0ru** [comment]
regarding the pasting of multi-lined text: it's possible to insert it when mediumjs is initialized with `Medium.richMode`, however this will also enable general html-content and pasted line-breaks are inserted as `<br>` per default, so we'll need some parsing to deal with these.

**Comment by m0ru** [comment]
The update also seems to have changed the produced html. Lines are no longer `<p>`s but `<div>`s for some reason.

**Comment by m0ru** [comment]
Nvm, that seems to be a Chrome vs FF issue. Chrome inserts `<div>`s to break lines, FF inserts `<br>`s

**Comment by m0ru** [comment]
Multi-line pasting whould work now. I tried dealing with all of contenteditable's quirks that managed to sneak through mediumjs.

**Pull request merged**